### DISCUSSION TITLE
TypeError: grammar is undefined

### DISCUSSION THREAD

**Discussion by AlphaBeta1906** [PrismJS/prism#3310] [Q&A] [answered]
Why i got this error?i trying to dynamically hiligth code in prism with this code :  
```javascript
Prism.highlight(paste.code, Prism.languages[paste.language], paste.language)
// paste.language = name of language i want to highlight
```
 it only worked for html,css,js and python(as far as i know),but when  try to hilight for example rust code i got error,i already try import all error languages but still not working except when i use it individually e.g : 
```javascript
Prism.highlight(paste.code, Prism.languages.rust, paste.language)
```
so how to solve  `TypeError: grammar is undefined` and use all languages in prism?
I use the mithril framework which uses vanilla javascript

**Comment by RunDevelopment** [comment, marked as answer]
A stack trace would have been nice. My guess is that the error occurs [here](https://github.com/PrismJS/prism/blob/bcb2e2c82ca86cbeeff435901be9f8ca35604163/components/prism-core.js#L692). This means that the languages isn't loaded.

It seems that you also changed something else when you changed `Prism.languages[paste.language]` -> `Prism.languages.rust`. Whatever that something was, it loaded all languages (or at least Rust) correctly.

While the incorrect language loading probably isn't Prism's fault, the very non-descriptive error message is. I'll add a better error. (see #3311)

---

Btw, the reason why HTML, CSS, and JS will always work is that those 3 languages (plus one util language) are always loaded by default. Prism was originally a very web-languages-focused highlighter.

**Reply by AlphaBeta1906** [reply]
So how to get prismjs to load all languages? and about the rust code that work i import rust component from prismjs(`import 'prismjs/components/prism-rust';`),but even though already import the language component i still cant use `Prism.languages[paste.language]`,so basically i need to use it individually.

**Reply by RunDevelopment** [reply]
That depends on your JS environment. [On NodeJS, you can use `loadLanguages()`](https://prismjs.com/#basic-usage-node) to load all languages. With webpack/any bundler, [our babel plugin](https://github.com/mAAdhaTTah/babel-plugin-prismjs) will handle loading all languages for you. Those are currently the only 2 ways to easily load all languages.

As a last resort, if none of those solutions fits your needs, then you could also download a bundle from our [download page](https://prismjs.com/download.html#themes=prism-tomorrow&languages=markup+css+clike+javascript).

**Reply by AlphaBeta1906** [reply]
Thanks,it works using babel plugin
